### PR TITLE
add `HintValueContext` for optimize hint.

### PR DESCRIPTION
**Pull request by dongzl**
For #20736 .

Changes proposed in this pull request:
  - add `HintValueContext` for optimize hint.

### CODE REVIEW

**Review by tuichenchuxin — COMMENTED** [review]
LGTM.

File: infra/binder/src/main/java/org/apache/shardingsphere/infra/binder/QueryContext.java [hunk 1]
```diff
@@ -38,10 +40,23 @@ public final class QueryContext {
     
     private String sqlStatementDatabaseName;
     
+    private HintValueContext hintValueContext;
+    
     public QueryContext(final SQLStatementContext<?> sqlStatementContext, final String sql, final List<Object> params) {
         this.sqlStatementContext = sqlStatementContext;
         this.sql = sql;
         parameters = params;
+        this.hintValueContext = SQLHintUtils.extractHint(sql);
```
Comment (tuichenchuxin) [inline infra/binder/src/main/java/org/apache/shardingsphere/infra/binder/QueryContext.java:49]: For this kind of untransferred, should we just not extract it?

**Review by tuichenchuxin — APPROVED** [review]
LGTM

**Comment by tuichenchuxin** [comment]
@strongduanmu Please have a look.

**Review by strongduanmu — APPROVED** [review]
Thank you for your excellent pr. 👍

**Pull request merged**